test/FunctionsV2/LongTimerTests.cs

```diff
@@ -103,5 +104,29 @@ public async Task EntitySignalWithLongDelay(bool extendedSessions)
                 await host.StopAsync();
             }
         }
+
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task WaitForExternalEventAboveMaxTime()
+        {
+            using (ITestHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.WaitForExternalEventAboveMaxTime),
+                enableExtendedSessions: false,
+                storageProviderType: "azure_storage",
+                durabilityProviderFactoryType: typeof(AzureStorageShortenedTimerDurabilityProviderFactory)))
+            {
+                await host.StartAsync();
+
+                var fireAt = TimeSpan.FromSeconds(90);
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.ApprovalWithTimeout), (fireAt, "throw"), this.output);
+                var status = await client.WaitForCompletionAsync(this.output, timeout: TimeSpan.FromMinutes(2));
+
+                Assert.Equal(OrchestrationRuntimeStatus.Completed, status?.RuntimeStatus);
+                Assert.Equal("TimeoutException", status?.Output);
+
+                await host.StopAsync();
+            }
+        }
```

Comment:
What is the "max time" being referenced in the method name? I don't think it's immediately clear from the name, it would also be good to add a short test-level comment explaining what this is testing for.

Reply:
Each storage provider can provide a method to validate timer lengths, as Azure Storage provider sets a max for 6 days. I improve the test name, but I could only make it so long before the taskhub name got too long for Azure Storage to support.